server/src/domain/asset/asset.service.ts

```diff
@@ -400,6 +398,44 @@ export class AssetService {
     return this.assetRepository.getAllByDeviceId(auth.user.id, deviceId);
   }
 
+  async get(auth: AuthDto, id: string): Promise<AssetResponseDto | SanitizedAssetResponseDto> {
+    await this.access.requirePermission(auth, Permission.ASSET_READ, id);
+
+    const asset = await this.assetRepository.getById(id, {
+      exifInfo: true,
+      tags: true,
+      sharedLinks: true,
+      smartInfo: true,
+      owner: true,
+      faces: {
+        person: true,
+      },
+      stack: {
+        exifInfo: true,
+      },
+    });
+
+    if (!asset) {
+      throw new BadRequestException('Asset not found');
+    }
+
+    if (!auth.sharedLink || auth.sharedLink?.showExif) {
+      const data = mapAsset(asset, { withStack: true });
+
+      if (data.ownerId !== auth.user.id) {
+        data.people = [];
+      }
+
+      if (auth.sharedLink) {
+        delete data.owner;
+      }
+
+      return data;
+    } else {
+      return mapAsset(asset, { stripMetadata: true, withStack: true });
+    }
```

Comment:
Can we flip this? So like `if (auth.SharedLink && !auth.sharedLink?.showExif) {return mapAsset(asset, { stripMetadata: true, withStack: true });}` and then all the other stuff below without an else. I prefer having as little deep code as possible

Reply:
Yeah, that makes sense. I copied it as is from the other one, but I like that approach better.